src/types/EmbeddedPaymentElement.tsx

```diff
@@ -0,0 +1,427 @@
+import {
+  requireNativeComponent,
+  NativeModules,
+  NativeEventEmitter,
+  EmitterSubscription,
+  LayoutAnimation,
+} from 'react-native';
+import type {
+  BillingDetails,
+  AddressDetails,
+  UserInterfaceStyle,
+  CardBrand,
+} from './Common';
+import type { PaymentMethod } from '.';
+import * as PaymentSheetTypes from './PaymentSheet';
+import type { ImageSourcePropType, ViewProps } from 'react-native';
+import NativeStripeSdk from '../NativeStripeSdk';
+import {
+  ReactElement,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import React from 'react';
+
+// Native bridge imports
+const { StripeSdk } = NativeModules;
+const NativeStripeEmbedded = StripeSdk;
+const eventEmitter = new NativeEventEmitter(NativeModules.StripeSdk);
+
+// -----------------------------------------------------------------------------
+// Types
+// -----------------------------------------------------------------------------
+
+/**
+ * The final result of a confirm call.
+ * Typically: payment succeeded (“completed”), canceled, or failed with error.
+ */
+export type EmbeddedPaymentElementResult =
+  | { status: 'completed' }
+  | { status: 'canceled' }
+  | { status: 'failed'; error: Error };
+
+/**
+ * The result of an update call.
+ */
+export type EmbeddedPaymentElementUpdateResult =
+  | { status: 'succeeded' }
+  | { status: 'canceled' }
+  | { status: 'failed'; error: Error };
+
+/**
+ * Contains details about a payment method that can be displayed to the customer in the embedded payment element UI.
+ */
+export interface PaymentOptionDisplayData {
+  /**
+   * An image representing the payment method, such as a VISA logo or Apple Pay icon.
+   */
+  image: ImageSourcePropType;
+  /**
+   * A user-facing label for the payment method, like "Apple Pay" or "•••• 4242" for a card.
+   */
+  label: string;
+  /**
+   * Optional billing details associated with the payment method, such as name, email, or address.
+   */
+  billingDetails?: BillingDetails;
+  /**
+   * A string identifier for the type of payment method.
+   * Stripe values: https://stripe.com/docs/api/payment_methods/object#payment_method_object-type
+   * External methods: https://stripe.com/docs/payments/external-payment-methods?platform=ios#available-external-payment-methods
+   * Apple Pay: "apple_pay"
+   */
+  paymentMethodType: string;
+  /**
+   * Optional base64-encoded regulatory mandate text.
+   * If `embeddedViewDisplaysMandateText = false`, this must be shown near the buy button in a tappable view that supports links.
+   */
+  mandateText?: string;
```

Comment:
Android mostly can't expose this as a string, because we have to be able to support links and such.

Reply:
What is it exposed as?

Reply:
I'd rather omit for now, and add later if someone needs it and make a call on it then.

Reply:
Yeah I think to support this we'd actually need to vend back a view w/ the mandate :(

Reply:
I'm opting to removing mandate stuff from this API, I think in RN merchants will need to let embedded show the mandate.